api/admin/controller/__init__.py

```diff
@@ -415,20 +417,29 @@ def _sort_by_collection(self, timestamps):
         collection ID and each value is a list of the timestamps associated with that collection."""
 
         result = {}
+        collection_name = None
         for timestamp in timestamps:
-            result.setdefault((timestamp.get("collection_id") or -1), []).append(timestamp)
+            result.setdefault(timestamp.get("collection_name"), []).append(timestamp)
         return result
 
     def _extract_info(self, timestamp):
         """Takes a Timestamp object and returns a dict"""
 
+        duration = None
+        if timestamp.start and timestamp.finish:
+            duration = (timestamp.finish - timestamp.start).total_seconds()
+
+        collection_name = "No associated collection"
+        if timestamp.collection_id:
+            collection_name = self._db.query(Collection).filter(Collection.id == timestamp.collection_id).one().name
+
```

Comment:
A simpler way to do this is: 

```
if timestamp.collection:
    collection_name = timestamp.collection.name
```

The SQL query will happen automatically.

In fact we can set this up so that the SQL queries aren't necessary at all, because `timestamp.collection` is populated from the database using the same query that gets all the `timestamp`s. I can show you how to do this on Monday. It's not actually necessary here because this controller will be used very rarely.